src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Summarize.cs

```diff
@@ -0,0 +1,245 @@
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+            Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
+
+            bool isValid = base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+            Contracts.Assert(returnType.IsTable);
+
+            // Source table type
+            DType sourceType = argTypes[0];
+
+            if (args[0] is AsNode)
+            {
+                isValid = false;
+                errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrSummarizeDataSourceScopeNotSupported);
+            }
+
+            if (sourceType.Contains(FunctionThisGroupScopeInfo.ThisGroup))
+            {
+                isValid = false;
+                errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrSummarizeDataSourceContainsThisGroupColumn);
+            }
+
+            var atLeastOneGroupByColumn = false;
+
+            for (int i = 1; i < args.Length; i++)
+            {
+                var argType = argTypes[i];
+                var arg = args[i];
+
+                DName columnName;
+                DType existingType;
+
+                // All args starting at index 1 need to be identifiers or aggregate functions (wrapped in AsNode node).
+                switch (arg)
+                {
+                    case AsNode nameNode:
+                        existingType = argType;
+                        columnName = nameNode.Right.Name;
+                        break;
+
+                    case FirstNameNode:
+                        if (!TryGetColumnLogicalName(sourceType, true, arg, errors, out columnName, out existingType))
+                        {
+                            isValid = false;
+                            continue;
+                        }
+
+                        if (DType.TryGetDisplayNameForColumn(sourceType, columnName, out var displayName))
+                        {
+                            if (displayName == FunctionThisGroupScopeInfo.ThisGroup)
+                            {
+                                isValid = false;
+                                errors.EnsureError(DocumentErrorSeverity.Severe, arg, TexlStrings.ErrSummarizeThisGroupColumnName);
+                                continue;
+                            }
+                        }
+
+                        atLeastOneGroupByColumn = true;
+                        break;
+
+                    default:
+                        isValid = false;
+                        errors.EnsureError(DocumentErrorSeverity.Severe, arg, TexlStrings.ErrSummarizeInvalidArg);
+                        continue;
+                }
+
+                if (columnName.Equals(FunctionThisGroupScopeInfo.ThisGroup))
+                {
+                    isValid = false;
```

Comment:
>isValid = false; [](http://example.com/codeflow?start=20&length=16)

Will this block ever be executed, or is it caught already on line 105?

Reply:
It will be executed.